### ISSUE TITLE
GPT2 train loss become nan since iter 3000

### ISSUE THREAD

**Issue by sampsonguo** [karpathy/nanoGPT#215]
using V100 32G
torchrun --standalone --nproc_per_node=1 train.py config/train_gpt2.py --dtype=float16

> 2023-03-22 09:30:39 iter 2920: loss 3.9883, time 1099.34ms, mfu 15.34%
> 
> 2023-03-22 09:30:50 iter 2930: loss 3.9408, time 1097.30ms, mfu 15.34%
> 
> 2023-03-22 09:31:01 iter 2940: loss 3.9994, time 1097.68ms, mfu 15.34%
> 
> 2023-03-22 09:31:12 iter 2950: loss 4.0977, time 1096.94ms, mfu 15.34%
> 
> 2023-03-22 09:31:23 iter 2960: loss 4.1535, time 1097.23ms, mfu 15.34%
> 
> 2023-03-22 09:31:34 iter 2970: loss 3.9453, time 1097.65ms, mfu 15.34%
> 
> 2023-03-22 09:31:45 iter 2980: loss 3.8993, time 1099.18ms, mfu 15.34%
> 
> 2023-03-22 09:31:56 iter 2990: loss 3.8853, time 1096.65ms, mfu 15.34%
> 
> 2023-03-22 09:32:32 step 3000: train loss 4.0390, val loss 4.0427
> 
> saving checkpoint to out
> 
> 2023-03-22 09:32:41 iter 3000: loss 4.2490, time 35392.33ms, mfu 13.85%
> 
> 2023-03-22 09:32:52 iter 3010: loss nan, time 1019.95ms, mfu 14.12%
> 
> 2023-03-22 09:33:02 iter 3020: loss nan, time 1015.95ms, mfu 14.36%
> 
> 2023-03-22 09:33:12 iter 3030: loss nan, time 1017.59ms, mfu 14.58%
> 
> 2023-03-22 09:33:22 iter 3040: loss nan, time 1016.57ms, mfu 14.78%
> 
> 2023-03-22 09:33:33 iter 3050: loss nan, time 1016.74ms, mfu 14.95%
> 
> 2023-03-22 09:33:43 iter 3060: loss nan, time 1018.23ms, mfu 15.11%
> 
> 2023-03-22 09:33:53 iter 3070: loss nan, time 1016.30ms, mfu 15.26%
> 
> 2023-03-22 09:34:03 iter 3080: loss nan, time 1015.92ms, mfu 15.39%
> 
> 2023-03-22 09:34:13 iter 3090: loss nan, time 1018.97ms, mfu 15.50%

**Comment by sampsonguo** [comment]
it seems problems solved. So close issue.
https://github.com/karpathy/nanoGPT/issues/167

**Comment by Baxkiller** [comment]
I got the same question and it occured since iter 16120.

> iter 15920: loss 3.0362, time 13112.75ms, mfu 10.26%
iter 15960: loss 2.9398, time 13114.33ms, mfu 10.26%
step 16000: train loss 3.1425, val loss 3.1335
saving checkpoint to out2
iter 16000: loss 3.1953, time 51375.24ms, mfu 9.50%
iter 16040: loss 3.1768, time 13123.16ms, mfu 9.58%
iter 16080: loss 3.1530, time 13119.77ms, mfu 9.65%
iter 16120: loss 3.1432, time 13120.89ms, mfu 9.71%
iter 16160: loss nan, time 12685.78ms, mfu 9.80%
iter 16200: loss nan, time 12685.91ms, mfu 9.88%
iter 16240: loss nan, time 12686.76ms, mfu 9.95%
iter 16280: loss nan, time 12686.53ms, mfu 10.02%

So I will appreciate if you can tell me the way you solved this problem. Thanks a lot.
